gramps/gen/lib/grampstype.py

```diff
@@ -151,6 +151,16 @@ def set_object_state(self, attr_dict):
         else:
             self.__string = ""
 
+    def __set_dict(self, value):
```

Comment:
Why use the double underscore? It isn't required.

Reply:
I was maintaining consistency with the existing `__set_*()` methods.
From the naming I assume that the original design intention was to call users to call `set()`, even where the type of `value` was already know. I'll address this as a separate PR

Reply:
See PR #1830